operator/pkg/ciliumidentity/pod.go

```diff
@@ -7,14 +7,24 @@ import (
 	"context"
 	"sync"
 
+	"github.com/cilium/cilium/operator/watchers"
 	"github.com/cilium/cilium/pkg/k8s/resource"
+	v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
 func podResourceKey(podName, podNamespace string) resource.Key {
 	return resource.Key{Name: podName, Namespace: podNamespace}
 }
 
+func isManagedByCilium(pod *v1.Pod) (bool, error) {
+	_, exists, err := watchers.HasCE(pod.Namespace, pod.Name)
```

Comment:
we need to handle the case when CiliumEndpoints are disabled and only CiliumEndpoitSlices are available and the other way around, basically, if CES are enabled we should check that, if not fallback to CE.